perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -845,7 +845,8 @@ public boolean isDirectGroupMember(PerunSession sess, Group group, Member member
 
 	@Override
 	public void addMember(PerunSession sess, List<Group> groups, Member member) throws WrongReferenceAttributeValueException, AlreadyMemberException, WrongAttributeValueException, GroupNotExistsException {
-		Collections.sort(groups, Collections.reverseOrder());
+		groups = new ArrayList<>(groups);
+		groups.sort(Collections.reverseOrder());
```

Comment:
Does sorting of input list breaks something or is it just some kind of precaution?

Reply:
Both. As a precaution, this should usually be done, instead of modifying the input objects, since its unspecified side-effect. Also, without this, this method cannot work with unmodifiable lists.

Reply:
OK. Thank you for explanation.